### ISSUE TITLE
DNA letter check logic

### ISSUE THREAD

**Issue by ftegenfe** [bbuchfink/diamond#756]
Hello,
when I run diamond on AA fasta files, it sometimes fails claiming there are only DNA letters even if that is not true.
Looking in the code I see this conforms with the logic.
This concerns the function SequenceFile::load_onepass() .
I wonder isn't it more logical to fail the check only if ALL sequences in the test range (CHECK_FOR_DNA_COUNT) are DNA?
I am aware of the --ignore-warning option but I encountered this in a package which uses diamond internally.
It's a suggestion - I fixed this by rearranging the sequences in the fasta file :)

**Comment by bbuchfink** [comment]
Yes that would probably make sense changing that, I'll look into it.

**Comment by bbuchfink** [comment]
It has been changed in the latest release.
